pkg/remote/resource_enumeration_error_handler.go

```diff
@@ -12,26 +12,51 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-type EnumerationAccessDeniedAlert struct {
-	message  string
-	provider string
+type ScanningPhase int
+
+const (
+	EnumerationPhase ScanningPhase = iota
+	DetailsFetchingPhase
+)
+
+type RemoteAccessDeniedAlert struct {
+	message       string
+	provider      string
+	scanningPhase ScanningPhase
 }
 
-func NewEnumerationAccessDeniedAlert(provider, supplierType, listedTypeError string) *EnumerationAccessDeniedAlert {
-	message := fmt.Sprintf("Ignoring %s from drift calculation: Listing %s is forbidden.", supplierType, listedTypeError)
-	return &EnumerationAccessDeniedAlert{message, provider}
+func NewRemoteAccessDeniedAlert(provider, supplierType, listedTypeError string, scanningPhase ScanningPhase) *RemoteAccessDeniedAlert {
+	var message string
+	switch scanningPhase {
+	case EnumerationPhase:
+		message = fmt.Sprintf("Ignoring %s from drift calculation: Listing %s is forbidden.", supplierType, listedTypeError)
+	case DetailsFetchingPhase:
+		message = fmt.Sprintf("Ignoring %s from drift calculation: Reading details of %s is forbidden.", supplierType, listedTypeError)
+	default:
+		message = fmt.Sprintf("Ignoring %s from drift calculation: %s", supplierType, listedTypeError)
+	}
+	return &RemoteAccessDeniedAlert{message, provider, scanningPhase}
 }
 
-func (e *EnumerationAccessDeniedAlert) Message() string {
+func (e *RemoteAccessDeniedAlert) Message() string {
 	return e.message
 }
 
-func (e *EnumerationAccessDeniedAlert) ShouldIgnoreResource() bool {
+func (e *RemoteAccessDeniedAlert) ShouldIgnoreResource() bool {
 	return true
 }
 
-func (e *EnumerationAccessDeniedAlert) GetProviderMessage() string {
-	message := "It seems that we got access denied exceptions while listing resources.\n"
+func (e *RemoteAccessDeniedAlert) GetProviderMessage() string {
+	var message string
+	switch e.scanningPhase {
+	case DetailsFetchingPhase:
+		message = "It seems that we got access denied exceptions while reading details of resources.\n"
+	case EnumerationPhase:
+		fallthrough
```

Comment:
It is strange to have a `fallthrough` to default which contains the message for enumeration error. Why not a simple if ?

```go
if e.scanningPhase == DetailsFetchingPhase {
  message = "It seems that we got access denied exceptions while reading details of resources.\n"
}
if e.scanningPhase == EnumerationPhase {
  message = "It seems that we got access denied exceptions while listing resources.\n"
}
```